go-controller/pkg/node/management-port_linux.go

```diff
@@ -21,34 +22,71 @@ const (
 // createPlatformManagementPort creates a management port attached to the node switch
 // that lets the node access its pods via their private IP address. This is used
 // for health checking and other management tasks.
-func createPlatformManagementPort(interfaceName, interfaceIP, routerIP, routerMAC string) error {
+func createPlatformManagementPort(interfaceName, interfaceIP, routerIP, routerMAC string,
+	initial bool) error {
+	var err error
+	var exists bool
+
 	link, err := util.LinkSetUp(interfaceName)
 	if err != nil {
 		return err
 	}
-	// Flush any existing IP addresses and assign the new IP
-	if err = util.LinkAddrFlush(link); err == nil {
-		err = util.LinkAddrAdd(link, interfaceIP)
+	if initial {
+		// for the initial setup we need to start from the clean slate, so flush
+		// all addresses on this link
+		err = util.LinkAddrFlush(link)
+		if err != nil {
+			return err
+		}
+	}
+	if exists, err = util.LinkAddrExist(link, interfaceIP); err == nil {
+		if !exists {
+			if !initial {
+				// we should log this so that one can debug as to why addresses are
+				// disappearing
+				klog.Warningf("missing IP address %s on the interface %s, adding it...",
+					interfaceIP, interfaceName)
+			}
+			err = util.LinkAddrAdd(link, interfaceIP)
+		}
 	}
 	if err != nil {
 		return err
 	}
 
+	// setup routes for the subnets that we care about
 	var allSubnets []string
 	for _, subnet := range config.Default.ClusterSubnets {
 		allSubnets = append(allSubnets, subnet.CIDR.String())
 	}
 	allSubnets = append(allSubnets, config.Kubernetes.ServiceCIDR)
-
-	// delete any existing routes and add new routes
-	if err = util.LinkRoutesDel(link, allSubnets); err == nil {
-		err = util.LinkRoutesAdd(link, routerIP, allSubnets)
+	if initial {
+		// start from a clean slate for the initial setup
+		err = util.LinkRoutesDel(link, allSubnets)
+		if err != nil {
+			return err
+		}
 	}
-	if err != nil {
-		if os.IsExist(err) {
-			klog.V(5).Infof("Ignoring error %s from 'route add %s via %s' - already added via IPv6 RA?",
-				err.Error(), config.Kubernetes.ServiceCIDR, routerIP)
-		} else {
+	for _, subnet := range allSubnets {
+		if exists, err := util.LinkRouteExists(link, routerIP, subnet); err == nil {
+			if exists {
+				continue
+			}
+			if !initial {
+				// we need to warn so that it can be debugged as to why routes are disappearing
+				klog.Warningf("missing route entry for subnet %s via gateway %s on link %v",
+					subnet, routerIP, link.Attrs().Name)
+			}
+			err = util.LinkRoutesAdd(link, routerIP, []string{subnet})
+			if err != nil {
+				if os.IsExist(err) {
+					klog.V(5).Infof("Ignoring error %s from 'route add %s via %s' - already added via IPv6 RA?",
+						err.Error(), config.Kubernetes.ServiceCIDR, routerIP)
+					continue
+				}
+			}
+		}
+		if err != nil {
```

Comment:
this err check seems unused? since you've got `if exists, err :=` the err inside the LinkRouteExists() block will be scoped only to that block. So this outer if err != nil will be checking an error from above the subnet for loop.

Reply:
Right, fixed it.